netx/httptransport/http3transport.go

```diff
@@ -27,6 +28,9 @@ func (t *HTTP3Transport) CloseIdleConnections() {
 func NewHTTP3Transport(config Config) RoundTripper {
 	txp := &HTTP3Transport{}
 	txp.QuicConfig = &quic.Config{}
+	if tlsdialer, ok := config.TLSDialer.(dialer.TLSDialer); ok {
+		txp.TLSClientConfig = tlsdialer.Config
+	}
```

Comment:
I would recommend to add a `TLSConfig *TLSConfig` field to `httptransport.Config` and modify the code to initialise it in `netx/netx.go` so that we don't need to obtain this field from the `TLSDialer`.